controller/transfer/project_transfer_manager.py

```diff
@@ -850,19 +880,23 @@ def import_file(
             )
 
     general.commit()
+    daemon.run(__post_processing_import_threaded, project_id, task_id, embedding_ids, data)
+
+
+def __post_processing_import_threaded(project_id: str, task_id: str,  embedding_ids: List[str], data: Dict[str, Any]) -> None:
+    time.sleep(5)
+    while True:
+        if task_queue.get_by_tokenization(project_id):
+            logger.info(f"Waiting for tokenization of project {project_id}")
+            time.sleep(5)
+        else:
+            logger.info(f"Tokenization finished, continue with embedding handling of project {project_id}")
+            break
```

Comment:
which this logic we now wait for tokenization for all embeddings. With the previous version (though faulty for token specific ones) we didn't wait.

Since only for token specific ones need the tokenization, I think we should do this in two parts (token embedding + waiting & attribute embedding + no waiting).

Reply:
Since this is not part of the task queue logic yet I would still keep it as this is so we do not run to much in parallel. As soon as we include our queue task manager here I would agree that we can resolve it by creating a dependency between different task types (ON_TOKEN Embeddings have an dependency on Tokenization; ON_ATTRIBUTE do not have it).

Reply:
Not changed